component/src/main/java/org/wso2/carbon/event/input/adapter/mqtt/internal/util/MQTTAdapterListener.java

```diff
@@ -0,0 +1,264 @@
+        this.connectionSSLEnabled = mqttBrokerConnectionConfiguration.isConnectionSSLEnabled();
+        this.sslTrustStoreLocation = MQTTEventAdapterConstants.ADAPTER_CONNECTION_SSL_TRUSTSTORE_LOCATION;
+        this.sslTrustStoreType = MQTTEventAdapterConstants.ADAPTER_CONNECTION_SSL_TRUSTSTORE_TYPE;
+        this.sslTrustStoreVersion = MQTTEventAdapterConstants.ADAPTER_CONNECTION_SSL_VERSION;
+        this.sslTrustStorePassword = MQTTEventAdapterConstants.ADAPTER_CONNECTION_SSL_TRUSTSTORE_PASSWORD;
+        this.topic = topic;
+        this.eventAdapterListener = inputEventAdapterListener;
+        this.tenantId = tenantId;
+        //Sorting messages until the server fetches them
+        String temp_directory = System.getProperty("java.io.tmpdir");
+        MqttDefaultFilePersistence dataStore = new MqttDefaultFilePersistence(temp_directory);
+
+        try {
+            // Construct the connection options object that contains connection parameters
+            // such as cleanSession and LWT
+            connectionOptions = new MqttConnectOptions();
+            connectionOptions.setCleanSession(cleanSession);
+            connectionOptions.setKeepAliveInterval(keepAlive);
+            //Create the secure connection
+            if (connectionSSLEnabled) {
+                char[] trustStorePassword = sslTrustStorePassword.toCharArray();
+                KeyStore keyStore = KeyStore.getInstance(sslTrustStoreType);
+                fileInputStream = new FileInputStream(sslTrustStoreLocation);
+                keyStore.load(fileInputStream, trustStorePassword);
+                TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance
+                        (TrustManagerFactory.getDefaultAlgorithm());
+                trustManagerFactory.init(keyStore);
+                SSLContext context = SSLContext.getInstance(sslTrustStoreVersion);
+                context.init(null, trustManagerFactory.getTrustManagers(), null);
+                connectionOptions.setSocketFactory(context.getSocketFactory());
+            }
+            if (this.mqttBrokerConnectionConfiguration.getBrokerPassword() != null) {
+                connectionOptions.setPassword(this.mqttBrokerConnectionConfiguration.getBrokerPassword().toCharArray());
+            }
+            if (this.mqttBrokerConnectionConfiguration.getBrokerUsername() != null) {
+                connectionOptions.setUserName(this.mqttBrokerConnectionConfiguration.getBrokerUsername());
+            }
+
+            // Construct an MQTT blocking mode client
+            mqttClient = new MqttClient(this.mqttBrokerConnectionConfiguration.getBrokerUrl(), this.mqttClientId,
+                                        dataStore);
+
+            // Set this wrapper as the callback handler
+            mqttClient.setCallback(this);
+        } catch (MqttException e) {
+            log.error("Exception occurred while subscribing to MQTT broker at "
```

Comment:
there’s a chance that the exception will be logged again further up the call stack, resulting in two log events for the same error.